docs/rfcs/1003-mandatory-human-handle.md

```diff
@@ -0,0 +1,205 @@
+
+```python
+def uncaseify(txt: str) -> str:
+    out = ""
+    for c in txt:
+        if c.isupper():
+            out += "_"
+            out += c
+        elif c == "_":
+            out += "__"
+        else:
+            out += c
+    return out
+
+assert uncaseify("a") == "a"
+assert uncaseify("_a_") == "__a__"
+assert uncaseify("aAa") == "a_aa"
+assert uncaseify("AA") == "_a_a"
+assert uncaseify("_A_") == "___a__"
+```
+
+`UserID` type has a hard limit of 32 bytes, while the email email address's limit
+is 255 bytes (including the domain part and `@`).
+The worst case for `uncaseify` is a user id only composed of uppercase (or underscores),
+in which case it size is doubled so 64 bytes.
+Considering domain is `redacted.invalid` (16 bytes long) and the `@`, there is
+238 bytes to accomodate the uncaseified user id (so far more than what is needed).
+
+Finally considering the unicode, the good news is there is nothing to do: our email
+format already accept international email address (so a regular user can already use
+something like `关羽@蜀.汉`).
+
+### 4 - `redacted.invalid` not allowed for real human handle
+
+Let's consider the following scenario:
+
+1 - User 1 (user id = "alice") is created with an old Parsec version, hence it has no human handle
+2 - User 2 (user id = "bob") is created with human handle email = "alice@redacted.invalid"
+
+Now we end up with Parsec displaying the same human handle email for both user 1 and 2 :(
+The easy solution for this is simply to consider `redacted.invalid` as a reserved domain that
+is not allowed for real human handle.
+
+Typically, `HumanHandle` method would look like:
+
+```rust
+impl HumanHandle {
+    pub fn new(email: &str, label: &str) -> Result<Self, &'static str> {
+      // Create regular human handle from the provided email & label
+      // here we ensure email doesn't use `redacted.invalid` domain
+    }
+
+    pub fn from_redacted_user(user_id: &UserID) -> Self {
+      // Here we compute label and email from the user id (hence email uses
+      // `redacted.invalid` domain)
+      // This method should only be used in the code deserializing `UserCertificate`
+    }
+```
+
+### 5 - Don't carry presence/absence of the fields to the GUI
+
+As we've seen, legacy certificates are handled just like redacted certificates. On top
+of that redacted certificates are only provided to Outsider user.
+
+Given legacy certificates are not that common, there is no need to convey this information
+to the GUI.
+
+Regarding redacted certificates, the GUI already has to know the user is an outsider
+to adapt what is displayed (e.g. outsider cannot create new workspaces).
+
+Hence there is no need to provide a specific field to inform where the human handle /
+device label fields come from.
+
+What this also means is a user can register himself with an email in the `redacted.invalid`
+tld. This is considered ok given the email is validated by the greeter before the creation.
```

Comment:
Isn't this phrase in contradiction with section 4 (`redacted.invalid not allowed for real human handle`)